crates/hc_sandbox/tests/cli.rs

```diff
@@ -769,6 +774,154 @@ async fn authorize_zome_call_credentials() {
         .stderr(Stdio::inherit())
         .kill_on_drop(true);
 
+    let mut hc_generate = input_piped_password(&mut cmd).await;
+
+    let launch_info = get_launch_info(&mut hc_generate).await;
+
+    let app_info = get_app_info(
+        launch_info.admin_port,
+        "test-app".into(),
+        *launch_info.app_ports.first().expect("No app ports found"),
+    )
+    .await;
+
+    let mut dna_hashes = match app_info {
+        AppResponse::AppInfo(Some(info)) => {
+            let cell_ids: Vec<Vec<CellInfo>> = info
+                .cell_info
+                .into_iter()
+                .map(|(_, cell_infos)| cell_infos)
+                .collect();
+            println!("cell_ids: {:?}", cell_ids);
+            let dna_hash_1 = match cell_ids[0].first().unwrap() {
+                CellInfo::Provisioned(cell) => cell.cell_id.dna_hash().clone(),
+                _ => panic!("Cell not provisioned"),
+            };
+            let dna_hash_2 = match cell_ids[1].first().unwrap() {
+                CellInfo::Provisioned(cell) => cell.cell_id.dna_hash().clone(),
+                _ => panic!("Cell not provisioned"),
+            };
+            let dna_hash_3 = match cell_ids[2].first().unwrap() {
+                CellInfo::Provisioned(cell) => cell.cell_id.dna_hash().clone(),
+                _ => panic!("Cell not provisioned"),
+            };
+            // The fixture happ contains 3 times the same dna of which 2 have the same dna hash
+            // therefore we need to deduplicate here...
+            vec![dna_hash_1, dna_hash_2, dna_hash_3]
+                .into_iter()
+                .collect::<HashSet<DnaHash>>()
+                .into_iter()
+                .collect::<Vec<DnaHash>>()
+        }
+        r => panic!("AppResponse does not contain app info: {:?}", r),
+    };
+
+    // ...and sort to get a consistent order to compare with output
+    dna_hashes.sort();
+
+    // Get agent meta info for all dnas
+    let mut cmd = get_sandbox_command();
+    cmd.env("RUST_BACKTRACE", "1")
+        .arg("--piped")
+        .arg(format!(
+            "--holochain-path={}",
+            get_holochain_bin_path().to_str().unwrap()
+        ))
+        .arg("call")
+        .arg("agent-meta-info")
+        .arg("--url")
+        .arg("wss://someurl:443")
+        .stdin(Stdio::piped())
+        .stdout(Stdio::piped())
+        .stderr(Stdio::inherit());
+
+    let mut hc_call = input_piped_password(&mut cmd).await;
+
+    hc_call.wait().await.unwrap();
+
+    let mut output = String::new();
+    hc_call
+        .stdout
+        .take()
+        .unwrap()
+        .read_to_string(&mut output)
+        .await
+        .unwrap();
+
+    let expected_output = format!(
+        r#"{{
+  "{}": {{}},
+  "{}": {{}}
+}}
+"#,
+        DnaHashB64::from(dna_hashes[1].clone()),
+        DnaHashB64::from(dna_hashes[0].clone())
```

Comment:
Both the DNA hashes and the space ids are sorted, so I wonder how this test could have passed? When running this locally, it fails consistently, and swapping the indexes fixes it consistently.

Reply:
> Since the wasm hashes are not deterministic I'm not sure how we can prevent this test from sporadically failing when we rebuild the fixture happ.

Right, that may vary, but the sorting algorithm is consistent and since space ids are sorted in the conductor method, and DNA hashes (which are the same with a prefix) are sorted in the test, it shouldn't matter whether the hashes are deterministic.